src/modules/ft/data_provider/history.rs

```diff
@@ -77,21 +87,21 @@ pub(crate) async fn get_ft_history(
 
         if db_info.status == "SUCCESS" {
             // TODO PHASE 2 this strange error will go away after we add absolute amounts to the DB
-            if (last_balance as i128) - delta < 0 {
+            if (last_balance.clone() - delta.clone()).is_negative() {
```

Comment:
This logic will go away when we switch to the new DB (it should happen asap), so I don't even try to make it cleaner

Reply:
@telezhnaya does BigDecimal implement comparison? It would be cleaner to just do

```suggestion
            if last_balance < delta {
```

Reply:
It should! Thanks